packages/client/README.md

```diff
@@ -10,97 +10,188 @@
 
-### Subscribing to real-time events in a stream
+```js
+const { StreamrClient } = require('streamr-client')
+```
+
+For usage in the browser include the latest build, e.g. by including a `<script>` tag pointing at a CDN:
+
+```html
+<!-- for Brubeck package (6x) -->
+<script src="https://unpkg.com/streamr-client@beta/streamr-client.web.js"></script>
+```
+___
+## Usage
+### Client Creation
+If you don't have an Ethereum account you can use the utility function [StreamrClient.generateEthereumAccount()](#utility-functions), which returns the address and private key of a fresh Ethereum account.
+
+More information on Stream IDs under the [stream creation project docs](https://streamr.network/docs/streams/creating-streams)
 
 ```js
-const sub = await client.subscribe({
-    stream: 'streamId',
-    partition: 0, // Optional, defaults to zero. Use for partitioned streams to select partition.
-    // optional resend options here
-}, (message, metadata) => {
-    // This is the message handler which gets called for every incoming message in the stream.
-    // Do something with the message here!
+const streamr = new StreamrClient({
+    auth: {
+        privateKey: 'your-private-key'
+    }
 })
 ```
 
-### Resending historical data
+Authenticating with an Ethereum private key contained in an Ethereum (web3) provider:
 
 ```js
-const sub = await client.resend(
-    'streamId',
-    {
-        last: 5,
-    }, 
-    (message) => {
-        // This is the message handler which gets called for every received message in the stream.
-        // Do something with the message here!
+const streamr = new StreamrClient({
+    auth: {
+        ethereum: window.ethereum,
     }
-)
+})
 ```
 
-See "Subscription options" for resend options
-
-### Programmatically creating a stream
+You can also create an anonymous client instance that will be allowed to interact with public streams:
+```js
+const streamr = new StreamrClient()
+```
 
+### Creating a stream 
 ```js
-const stream = await client.createStream({
-    id: '/foo/bar', // or 0x1234567890123456789012345678901234567890/foo/bar or mydomain.eth/foo/bar
+// Requires MATIC tokens (Polygon blockchain gas token)
+const stream = await streamr.createStream({
+    id: '/foo/bar'
 })
-console.log(`Stream ${stream.id} has been created!`)
 
-// Optional: to enable historical data resends, add the stream to a storage node
-const { StorageNode, StreamrClient } = require('streamr-client').StreamrClient
-await stream.addToStorageNode(StorageNode.STREAMR_GERMANY)
+console.log(stream.id) // `${address}/foo/bar`
+```
 
-// Do something with the stream, for example call stream.publish(message)
+### Subscribing to real-time events in a stream
+The callback's first parameter, `payload`, will contain the value given to the `publish` method. The second parameter `streamrMessage` is of type StreamrObject. [You can read more about it here](../protocol/src/protocol/message_layer/StreamMessage.ts)
```

Comment:
Maybe the metadata can be explained in a separate paragraph, as it is a bit more advanced usage?